### ISSUE TITLE
marrie does not play the latest episode

### ISSUE THREAD

**Issue by rohshall** [rafaelmartins/marrie#6]
```
$ marrie -v
marrie 0.4
```

```
$ marrie -l
Podcast feeds available:

    upfirst - https://feeds.npr.org/510318/podcast.xml
```

```
$ marrie -g upfirst
Fetching the latest chapter available for "upfirst"

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   357  100   357    0     0   1330      0 --:--:-- --:--:-- --:--:--  1332
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 11.6M  100 11.6M    0     0  1710k      0  0:00:07  0:00:07 --:--:-- 2120k
```

```
17:24 salil@dell:~/marrie$ marrie -p upfirst
Playing the latest chapter fetched for "upfirst"

src/streamdump.c:stream_open():600] error: failed to open file: /home/salil/podcasts/upfirst/20230602_upfirst_a42efb5a-707e-4bb2-8626-6ffb782063e0.mp3?d=771: No such file or directory
```

I added some debug statements in `def _fetch(self, url)` and it prints this:

`Fetching https://play.podtrac.com/npr-510318/edge1.pod.npr.org/anon.npr-mp3/npr/upfirst/2023/06/20230602_upfirst_a42efb5a-707e-4bb2-8626-6ffb782063e0.mp3?d=771&size=12264639&e=1179612846&t=podcast&p=510318&awEpisodeId=1179612846&awCollectionId=510318 which resolves to 20230602_upfirst_a42efb5a-707e-4bb2-8626-6ffb782063e0.mp3?d=771&size=12264639&e=1179612846&t=podcast&p=510318&awEpisodeId=1179612846&awCollectionId=510318`

So, basically, when we use posixpath on the url, the filename and therefore the filepath is not correct. So, `_latest_file` is set to the filepath with query params, and it fails to play it.

I also noticed that `.latest` symbolic link is not getting created.
